jmx-metrics/src/test/groovy/io/opentelemetry/contrib/jmxmetrics/target-systems/CassandraIntegrationTests.groovy

```diff
@@ -31,170 +30,164 @@ class CassandraIntegrationTests extends OtlpIntegrationTest  {
         Testcontainers.exposeHostPorts(otlpPort)
         configureContainers('target-systems/cassandra.properties',  otlpPort, 0, false)
 
-        expect:
-        when: 'we receive metrics from the JMX metrics gatherer'
-        List<ResourceMetrics> receivedMetrics = collector.receivedMetrics
-        then: 'they are of the expected size'
-        receivedMetrics.size() == 1
+        ArrayList<Metric> metrics
+        await().atMost(30, TimeUnit.SECONDS).untilAsserted {
+            List<ResourceMetrics> receivedMetrics = collector.receivedMetrics
+            assert receivedMetrics.size() == 1
 
-        when: "we examine the received metric's instrumentation library metrics lists"
-        ResourceMetrics receivedMetric = receivedMetrics.get(0)
-        List<InstrumentationLibraryMetrics> ilMetrics =
-                receivedMetric.instrumentationLibraryMetricsList
-        then: 'they of the expected size'
-        ilMetrics.size() == 1
+            ResourceMetrics receivedMetric = receivedMetrics.get(0)
+            List<InstrumentationLibraryMetrics> ilMetrics =
+                    receivedMetric.instrumentationLibraryMetricsList
+            assert ilMetrics.size() == 1
 
-        when: 'we examine the instrumentation library'
-        InstrumentationLibraryMetrics ilMetric = ilMetrics.get(0)
-        InstrumentationLibrary il = ilMetric.instrumentationLibrary
-        then: 'it is of the expected content'
-        il.name  == 'io.opentelemetry.contrib.jmxmetrics'
-        il.version == '1.0.0-alpha'
+            InstrumentationLibraryMetrics ilMetric = ilMetrics.get(0)
+            InstrumentationLibrary il = ilMetric.instrumentationLibrary
+            assert il.name  == 'io.opentelemetry.contrib.jmxmetrics'
+            assert il.version == '1.0.0-alpha'
```

Comment:
I think this should be updated so untilAsserted may be silencing the failures:

```
NOTE: Be VERY careful so that you're not using this method incorrectly in languages (like Kotlin and Groovy) that doesn't disambiguate between a ThrowingRunnable that doesn't return anything (void) and Callable that returns a value. For example in Kotlin you can do like this:
 
       await().untilAsserted { true == false }
       
and the compiler won't complain with an error (as is the case in Java). If you were to execute this test in Kotlin it'll pass!
```

Reply:
I'm sorry, forgot this wasn't set by the sdk: https://github.com/open-telemetry/opentelemetry-java-contrib/blob/4bfd97dfff1ad7d58898a9e0aa2a43ce82794f3a/jmx-metrics/src/main/java/io/opentelemetry/contrib/jmxmetrics/GroovyMetricEnvironment.java#L102

Reply:
I can take care these when adopting Awaitility.await in another PR.

Reply:
Ah hehe yeah let's do in another PR then best would be to fill the version into a file and read it similar to what the SDK does